editoast/src/views/timetable/occupancy_block.rs

```diff
@@ -0,0 +1,189 @@
+use editoast_models::DbConnection;
+use editoast_schemas::primitives::Identifier;
+use itertools::izip;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::sync::Arc;
+use tracing::info;
+use utoipa::ToSchema;
+
+use crate::ValkeyClient;
+use crate::ValkeyConnection;
+use crate::core::AsCoreRequest;
+use crate::core::CoreClient;
+use crate::core::pathfinding::TrackRange;
+use crate::core::signal_projection::SignalUpdate;
+use crate::core::signal_projection::SignalUpdatesRequest;
+use crate::core::signal_projection::TrainSimulation;
+use crate::error::Result;
+use crate::models;
+use crate::models::infra::Infra;
+use crate::models::prelude::*;
+use crate::views::projection::ProjectPathInput;
+use crate::views::projection::TrainSimulationDetails;
+use crate::views::projection::extract_train_details;
+use crate::views::timetable::simulation::train_simulation_batch;
+use crate::views::timetable::train_schedule::TrainScheduleError;
+
+/// Occupancy block output is described by time-space points and blocks
+#[derive(Debug, Clone, Deserialize, Serialize, ToSchema)]
+pub struct OccupancyBlockResult {
```

Comment:
```suggestion
pub struct OccupancyBlocks {
```

`Result` is kinda coined by Rust (I know we use it extensively, but...). If you prefer having a suffix anyway, what about `-Response`?

Reply:
done